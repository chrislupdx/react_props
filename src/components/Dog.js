import React from 'react';
import propTypes from 'prop-types';

function Dog({ name, age, weight }) {
  return (
    <dl>
      <dt>Name</dt>
      <dd>{name}</dd>

      <dt>Age</dt>
      <dd>{age}</dd>

      <dt>Weight</dt>
      <dd>{weight ? weight : 'unknown' }</dd>
    </dl>
  );
}

Dog.propTypes = {
  name: propTypes.string.isRequired,
  age: propTypes.number.isRequired,
  weight: propTypes.string.isRequired
    
};

export default Dog;
