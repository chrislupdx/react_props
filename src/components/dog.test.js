import React from 'react';
import { shallow } from 'enzyme';
import Dog from './Dog';

describe('Dog component', () => {
  it('renders dog info', () => {
    const wrapper = shallow(<Dog name= "spot" age={4} weight= "1" />);
    expect(wrapper).toMatchSnapshot();
  });
});
