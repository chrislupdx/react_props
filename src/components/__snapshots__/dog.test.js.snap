// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dog component renders dog info 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    spot
  </dd>
  <dt>
    Age
  </dt>
  <dd>
    4
  </dd>
  <dt>
    Weight
  </dt>
  <dd>
    1
  </dd>
</dl>
`;
